### PR TITLE
migration: Fixed all instance of bad migrations and got tf-scalar-dashboard to be functional.

### PR DESCRIPTION
**Pull request by stephanwlee**
With this commit, `bazel run tensorboard -- --logdir foo` now shows functional scalar dashboard with:
- faster incremental build (<6s)
- source maps (not to TypeScript, yet)

cc @bileschi

### CODE REVIEW

**Comment by wchargin** [comment]
This is awesome—thanks for putting in the extra effort to get this
working incrementally!

File: tensorboard/components/tf_tensorboard/tf-tensorboard.ts [hunk 1]
```diff
@@ -0,0 +1,1073 @@
...
+    type: Object,
+  })
+  _canceller = new Canceller();
+
+  @property({
+    type: Boolean,
+  })
+  _refreshing: boolean = false;
+
+  // behaviors: [AutoReloadBehavior];
```
Comment (psybuzz) [inline tensorboard/components/tf_tensorboard/tf-tensorboard.ts]: drop

File: tensorboard/components/tf_tensorboard/tf-tensorboard.ts [hunk 2]
```diff
@@ -0,0 +1,1073 @@
...
+  environmentStore,
+  runsStore,
+  experimentsStore,
+  Router,
+} from '../tf_backend';
+import '../tf_dashboard_common/tensorboard-color';
+import {setUseHash} from '../tf_globals/globals';
+import {getLimit, setLimit} from '../tf_paginated_view/paginatedViewStore';
+import {getString, setString, TAB} from '../tf_storage/storage';
+// import { DO_NOT_SUBMIT } from "../tf-plugin-util/plugin-host.html";
```
Comment (psybuzz) [inline tensorboard/components/tf_tensorboard/tf-tensorboard.ts]: should now be able to import the real module

File: tensorboard/components/tf_tensorboard/plugin-dialog.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (psybuzz) [inline tensorboard/components/tf_tensorboard/plugin-dialog.ts]: These should now all be 2020 :D

File: tensorboard/components/tf_tensorboard/plugin-dialog.html [hunk 4]
```diff
@@ -1,114 +0,0 @@
...
-        },
-        _hidden: {
-          type: Boolean,
-          computed: '_computeHidden(_open)',
-          reflectToAttribute: true, // for CSS
-        },
-        // Suppress the native paper-dialog backdrop. We use a custom one. We make
-        // the dialog reference this bound property instead of directly setting
-        // the attribute to the string "false" in the HTML because Polymer
-        // unfortunately parses "false" as a true-ey value (a non-empty string).
```
Comment (psybuzz) [inline tensorboard/components/tf_tensorboard/plugin-dialog.html]: Let's preserve this comment, seems possibly useful

File: tensorboard/components/tf_tensorboard/BUILD [hunk 5]
```diff
@@ -1,85 +1,63 @@
...
-        "@org_polymer_iron_icons",
-        "@org_polymer_paper_button",
-        "@org_polymer_paper_checkbox",
-        "@org_polymer_paper_dialog",
-        "@org_polymer_paper_header_panel",
-        "@org_polymer_paper_icon_button",
-        "@org_polymer_paper_listbox",
-        "@org_polymer_paper_tabs",
-        "@org_polymer_paper_toolbar",
+        # "@com_google_fonts_roboto",
```
Comment (psybuzz) [inline tensorboard/components/tf_tensorboard/BUILD]: Can this be dropped / Do we have a plan to re-add it?

File: tensorboard/components/polymer_config.ts [hunk 6]
```diff
@@ -13,6 +11,5 @@
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
--->
-
-<script src="registry.js"></script>
+==============================================================================*/
+(window as any).polymerSkipLoadingFontRoboto = true;
```
Comment (psybuzz) [inline tensorboard/components/polymer_config.ts]: Who is loading Roboto now?

File: tensorboard/components/index.html [hunk 7]
```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
```
Comment (psybuzz) [inline tensorboard/components/index.html]: With this, we can delete the `tensorboard.html` file in the same folder?

**Comment by psybuzz** [comment]
Thanks for the effort, functionaltests/ pass too!  I am a little worried that between fixing tests, copybara, etc there could be enough edge cases to make us revisit Scalars multiple times.  Or maybe that's overthinking things.

A couple questions:

The font used in the x/y axes in the line charts is different (less bold).

Filtering the Runs list does not work.  Is that related to these changes, or with RunsSelector migration?

On load, Console shows errors that seem to be related to D3 overwriting the global `dispatchEvent()`, which interferes with the Polymer LegacyElementMixin.  Can we fix this or at least check that UI events aren't broken?

image

File: tensorboard/defs/rollup.config.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+const commonjs = require('rollup-plugin-commonjs');
+
+module.exports = {
+  plugins: [
+    node({
+      mainFields: ['browser', 'es2015', 'module', 'jsnext:main', 'main'],
+    }),
+    commonjs({
+      // cjs and rollup are incapable of understanding Lodash's exports.
+      // https://github.com/rollup/rollup-plugin-commonjs/issues/266
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/defs/rollup.config.js]: Unclear whether tree shaking will properly operate. Will keep this for now though.

File: tensorboard/components/polymer_config.ts (hunk 6 above)
Comment (stephanwlee) [reply tensorboard/components/polymer_config.ts]: This is loading one from CDN and it is a behavior change we did not intend. 

We should load the Roboto from our own backend instead.

**Comment by stephanwlee** [comment]
About the font used in plottable, it looks like Plottable removed the default `Helvetica Neue` on the axis (https://github.com/palantir/plottable/pull/3447).

File: tensorboard/components/vz_chart_helpers/plottable-interactions.ts [hunk 9]
```diff
@@ -216,6 +212,11 @@ class TouchDispatcher extends (Plottable.Dispatchers.Touch as any) {
 }
 
 export class PointerInteraction extends Plottable.Interactions.Pointer {
+  static preallocateTbDispatcher(component: Plottable.Component) {
+    MouseDispatcher.getDispatcher(component);
+    TouchDispatcher.getDispatcher(component);
+  }
```
[the remark this answers is not in the document]
Comment (psybuzz) [reply tensorboard/components/vz_chart_helpers/plottable-interactions.ts]: Explained offline.  It might be good to have a short comment along the lines: "allocate our custom Dispatchers first to skip Plottable's default ones".  I leave it up to you.

File: tensorboard/components/analytics.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+(window as any)['ga'] = function() {};
```
Comment (psybuzz) [inline tensorboard/components/analytics.ts]: License
